### ISSUE TITLE
Cacheable lookup responds with ESERVFAIL while native dns.resolve works

### ISSUE THREAD

**Issue by lokesh411** [szmarczak/cacheable-lookup#68]
Hello,
I have come across a issue where cacheable.lookup or cacheable.lookupAsync was giving `ESERVFAIL` while dns.resolve or dns.lookup worked perfectly fine.
When i went through the code, i found out that we are trying to resolve for both IPv4 and IPv6 addresses for the domain. If the server sends failure (ESERVFAIL) for the dns lookup since the hostname doesn't exists in the respective IP family, even if it can resolve for the other IP family the library throws the error.
Reference:
[https://github.com/szmarczak/cacheable-lookup/blob/master/source/index.js#L244]
Any reasons where we should lookup for both IPv4 and IPv6 addresses?
`const [A, AAAA] = await Promise.all([
			ignoreNoResultErrors(this._resolve4(hostname, ttl)),
			ignoreNoResultErrors(this._resolve6(hostname, ttl))
]);`
For a domain i found that resolve4 was working fine but resolve6 was failing with `ESERVFAIL` error

**Comment by vinaycr7** [comment]
I have also faced the same issue

**Comment by szmarczak** [comment]
Duplicate of #52. What does `ESERVFAIL` mean? I couldn't find anything in the internet about this.

**Comment by lokesh411** [comment]
https://bluecatnetworks.com/blog/the-top-four-dns-response-codes-and-what-they-mean/
@szmarczak This explains all the the DNS errors, `ESERVFAIL` means that the query failed because the answer cannot be given, means the server refuses to give response for the query. 
Shouldn't be handled in a way where we would get only one query IPv4 and if the DNS is not present we could call IPv6 right?

**Comment by szmarczak** [comment]
The website doesn't work for me. Can you post a screenshot or give another source please?

**Comment by lokesh411** [comment]
<img width="848" alt="Screenshot 2022-09-08 at 7 34 28 PM" src="https://user-images.githubusercontent.com/26621756/189142877-7b2b3dc0-7630-45e5-b3f0-eef72d2a6d64.png">
[](url)
<img width="761" alt="Screenshot 2022-09-08 at 7 35 09 PM" src="https://user-images.githubusercontent.com/26621756/189143013-a29c727e-728f-49f0-a1cf-446c03ee8eb7.png">

**Comment by lokesh411** [comment]
Yeah in some DNS servers this happens, when i point to CF dns it returns with `ENOTFOUND`. But we don't have to query for IPv4 and IPv6 addresses right on resolve()

**Comment by szmarczak** [comment]
Correct, that's part of #52

**Comment by lokesh411** [comment]
The problem still exists, when i query for some domains like staging.camsrepository.com, CF DNS resolver throws ESERVFAIL. Since the implementation is querying for both resolve4 and resolve6, should we use promise.allSettled instead of promise.all
Code reference: https://github.com/szmarczak/cacheable-lookup/blob/master/source/index.mjs#L251

**Comment by SuperOleg39** [comment]
Hello! Is this temporary workaround correct?
          
```
        const cacheable = new CacheableLookup({
            cache,
            maxTtl,
          });

          const originalLookup = cacheable.lookup;

          cacheable.lookup = (hostname, options, callback) => {
            if (hostname === 'localhost') {
              return dns.lookup(hostname, options, callback);
            }
            originalLookup.call(cacheable, hostname, options, callback);
          };

          cacheable.install(http.globalAgent);
          cacheable.install(https.globalAgent);
```
